Express notes api - YoonHwan Cho

YoonHCho opened the pull request:
Express notes api - YoonHwan Cho
![exress notes api](https://user-images.githubusercontent.com/106636383/186070734-8ba6390b-6e2f-4bf9-8861-2aad41d861da.gif)

TimelikeClosure commented on express-notes-api/index.js:
```diff
@@ -0,0 +1,97 @@
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNum = Number(req.params.id);
+  if (idNum < 0 || isNaN(idNum)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
```
- Ids should also be an _integer_ (not just a number) in addition to being positive.
- `0` doesn't count as positive.

TimelikeClosure commented on express-notes-api/index.js:
```diff
@@ -0,0 +1,97 @@
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      } else {
+        res.sendStatus(200);
```
A successful update should also include the updated note in the response:

---

![image](https://user-images.githubusercontent.com/16143771/186225390-3fb34a43-1cbf-42f0-9348-84f31862253f.png)

YoonHCho commented:
changed to check if less or equal to 0 and if not a number/integer
`if (idNum <= 0 || !Number.isInteger(idNum)) {`

and updated to response with status and the updated like on the example picture
`res.status(200).json(dataJSON.notes[idNum]);`
